control-plane/roles/auditing-meili/tasks/main.yaml

```diff
@@ -19,18 +19,30 @@
       kind: Secret
       metadata:
         name: metal-auditing-master-key
-        namespace: "{{ metal_control_plane_namespace }}"
+        namespace: "{{ metal_auditing_namespace }}"
       stringData:
         MEILI_MASTER_KEY: "{{ auditing_meili_secret }}"
   when: auditing_meili_secret
 
+- name: Create meili secret for control plane
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: metal-auditing-master-key
+        namespace: "{{ metal_control_plane_namespace }}"
+      stringData:
+        MEILI_MASTER_KEY: "{{ auditing_meili_secret }}"
+  when: auditing_meili_secret and (metal_control_plane_namespace != metal_auditing_namespace)
```

Comment:
This appears a little too specific to me / it is introducing some kind of a role coupling.

My suggestion would be not to use `metal_auditing_meili_secret_name` in the `metal` role. Instead use `metal_auditing_meili_api_key` and do this in the defaults:

```
metal_auditing_meili_api_key: "{{ lookup('k8s', api_version='v1', namespace=metal_auditing_namespace, kind='Secret', resource_name='metal-auditing-master-key').get('data', {}).get('MEILI_MASTER_KEY') | b64decode if metal_auditing_enabled else '' }}"
```

This is already done by the headscale role, too. This will make your role a bit more self-contained. :)

Reply:
Thanks for the suggestion!